base/abstractarray.jl

```diff
@@ -625,6 +625,13 @@ convert{T,S,N}(::Type{AbstractArray{T  }}, A::AbstractArray{S,N}) = convert(Abst
 
 convert{T,N}(::Type{Array}, A::AbstractArray{T,N}) = convert(Array{T,N}, A)
 
+"""
+   of_indices(x, y)
+
+Represents the array `y` as an array having the same indices type as `x`.
+"""
+of_indices(x, y) = similar(dims->y, oftype(indices(x), indices(y)))
```

Comment:
It's basically a test to see if this is something generally useful. Since it's not exported, isn't that reasonable? 

I could delete the docstring if you think it would be better.